app/master/build_scheduler.py

```diff
@@ -101,18 +99,23 @@ def execute_next_subjob_or_free_executor(self, slave):
             # this method, finds the subjob queue empty, and is torn down.  If that was the last 'living' slave, the
             # build would be stuck.
             with self._subjob_assignment_lock:
+                is_first_subjob = False
+                if self._build._unstarted_subjobs.qsize() == len(self._build.all_subjobs()):
+                    is_first_subjob = True
                 subjob = self._build._unstarted_subjobs.get(block=False)
                 self._logger.debug('Sending subjob {} (build {}) to slave {}.',
                                    subjob.subjob_id(), subjob.build_id(), slave.url)
                 try:
                     slave.start_subjob(subjob)
                     subjob.mark_in_progress(slave)
-
                 except SlaveMarkedForShutdownError:
                     self._build._unstarted_subjobs.put(subjob)  # todo: This changes subjob execution order. (Issue #226)
                     # An executor is currently allocated for this subjob in begin_subjob_executions_on_slave.
                     # Since the slave has been marked for shutdown, we need to free the executor.
                     self._free_slave_executor(slave)
+                else:
+                    if is_first_subjob:
+                        self._build.mark_started()
```

Comment:
Yeah, I changed this behavior a little bit. If I didn't change this, and instead kept it as it was, then _build.mark_started() would get called multiple times, and lead to a FSM state transition error: 

"Error during build state transition. (FysomError: event BuildEvent.START_BUILDING inappropriate in current state BuildState.BUILDING)".

Alternatively, I could have added a build transition from the BUILDING state to itself upon multiple START_BUILDING events and left the self._build.mark_started() in the allocate_slave() method, but I thought it'd make more sense to not mark a build started multiple times. I'm open to keeping it the way it was.